app/src/main/java/com/anysoftkeyboard/AnySoftKeyboard.java

```diff
@@ -1350,9 +1349,12 @@ private void onFunctionKey(final int primaryCode, final Key key, final int multi
                 if (mVoiceRecognitionTrigger.isInstalled()) {
                     mVoiceRecognitionTrigger.startVoiceRecognition(getCurrentAlphabetKeyboard().getDefaultDictionaryLocale());
                 } else {
-                    Intent voiceInputNotInstalledIntent = new Intent(getApplicationContext(), VoiceInputNotInstalledActivity.class);
-                    voiceInputNotInstalledIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                    startActivity(voiceInputNotInstalledIntent);
+                    Context context = getApplicationContext();
```

Comment:
I think that this will be very confusing for non-power-users:
They will get a _toast_ and will not know what to do to get voice functionality.

This is the main reason that Activity exists - it give information and have a button that direct the user to the Play-Store to download the voice-input app.